### PR TITLE
When using a custom compositor, ensure the root canvas is flushed.

### PR DESCRIPTION
**Pull request by chinmaygarde**
The root canvas is managed by the external view embedder when using a custom
compositor. Due to this, frame submission on the surface will not end up
flushing the same (because the surface doesn’t have it to begin with). Fixed
with tests.

Fixes part 1 of b/139737705 and adds a disabled test for part 2.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
cc @dnicoara: This fixes the blocking part of the issue filed. I should have the second non-blocking issue described in that ticket fixed in a few minutes but figured you might want to unblock yourself ASAP.

File: shell/common/rasterizer.cc [hunk 1]
```diff
@@ -259,6 +259,10 @@ RasterStatus Rasterizer::DrawToSurface(flutter::LayerTree& layer_tree) {
     }
     frame->Submit();
     if (external_view_embedder != nullptr) {
+      // Frame submission will only flush the surface canvas. But we are
+      // rendering into a canvas managed by the external view embedder. This
+      // must be flush manually.
```
Comment (jason-simmons) [inline shell/common/rasterizer.cc]: typo: should be "flushed"

File: shell/common/rasterizer.cc [hunk 2]
```diff
@@ -259,6 +259,10 @@ RasterStatus Rasterizer::DrawToSurface(flutter::LayerTree& layer_tree) {
     }
     frame->Submit();
     if (external_view_embedder != nullptr) {
+      // Frame submission will only flush the surface canvas. But we are
+      // rendering into a canvas managed by the external view embedder. This
+      // must be flushed manually.
+      root_surface_canvas->flush();
```
Comment (amirh) [inline shell/common/rasterizer.cc]: As discussed - move this to EmbedderExternalViewEmbedder::SubmitFrame to avoid double flush on iOS

Comment (chinmaygarde) [reply shell/common/rasterizer.cc]: Done.

**Pull request merged**